src/components/Select/Select.jsx

```diff
@@ -0,0 +1,220 @@
+    const [filled, setFilled] = useState(false)
+    const [optionSelected, setOptionSelected] = useState()
+    const containerRef = useRef(null)
+
+    const handleChange = option => {
+      setOptionSelected(option)
+      setFilled(true)
+      setFocus(false)
+    }
+
+    const outsideAlerter = ref => {
+      useEffect(() => {
+        const handleClickOutside = event => {
+          ref.current && !ref.current.contains(event.target) && setFocus(false)
+        }
+
+        document.addEventListener('mousedown', handleClickOutside)
+        return () => {
+          document.removeEventListener('mousedown', handleClickOutside)
+        }
+      }, [ref])
+    }
+
+    outsideAlerter(containerRef)
+
+    return (
+      <Wrapper ref={containerRef} disabled={disabled} error={error} focus={focus} quiet={quiet} {...props}>
+        {!quiet && <Label>{label}</Label>}
+        <Container tabIndex='0'>
+          <SelectContainer
+            className='selectContainer'
+            focus={focus}
+            quiet={quiet}
+            error={error}
+            disabled={disabled}
+            onClick={() => !disabled && setFocus(!focus)}
+          >
+            <SelectBase tabIndex='-1' value={optionSelected} onChange={handleChange} filled={filled} ref={ref}>
+              {disabled ? (
+                <option value=''>{optionLabel}</option>
+              ) : (
+                options.map(option => (
+                  <option key={option} value={option[optionValue]}>
+                    {optionSelected || optionLabel}
+                  </option>
+                ))
+              )}
+            </SelectBase>
+            {!disabled && focus ? <ExpandLess /> : <ExpandMore />}
+          </SelectContainer>
+
+          {focus && (
+            <OptionsContainer>
+              {options.map(option => (
+                <div key={option} onClick={() => handleChange(option[optionValue])}>
+                  <OptionContainer>
+                    <OptionLabel>{option[optionLabel]}</OptionLabel>
+                    {option[optionValue] === optionSelected && <IconCheck />}
+                  </OptionContainer>
+                </div>
+              ))}
+            </OptionsContainer>
+          )}
+        </Container>
+        {!focus && <Message>{caption}</Message>}
+      </Wrapper>
+    )
+  }
+)
+
+const errorVariant = variant({
+  prop: 'error',
+  default: false,
+  variants: {
+    true: css`
+      p {
+        color: error;
+      }
+      .selectContainer {
+        border-style: solid;
+        border-color: error;
+        select {
+          color: gray.900;
+        }
+      }
+    `,
+    false: css``
+  }
+})
+
+const disabledVariant = variant({
+  prop: 'disabled',
+  default: false,
+  variants: {
+    true: css`
+      p {
+        color: disabled;
+      }
+      div {
+        cursor: default;
+        border-color: gray.400;
+        background-color: ${({ quiet }) => (quiet ? '#fff' : '#F5F5F5')};
```

Comment:
como usamos o xstyled tu pode passar o nome correspondente a cor aqui
```suggestion
        background-color: ${({ quiet }) => (quiet ? 'white' : 'gray.100')};
```

Reply:
Foi como tentei inicialmente, mas exatamente neste tipo de declaração onde utilizo um if ternário e um $ para passar a cor, ele não pega a referência para as cores, como o "gray.100". Só consegui resolver enviando a cor diretamente.

Reply:
cara realmente não sei, talvez pegar o theme nessa função seja uma saida, vou dar uma olhada na doc do xstyled pra ver se tem algo tb. Mas é mal usar a cor direto ai pq caso a paleta de cores seja alterada em algum momento não precisamos passar em todos componentes e alterar eles separadamente

Reply:
Acredito que tu precisa disso https://xstyled.dev/docs/system-utilities/#theme-getters.